cmd/admin-handlers-users.go

```diff
@@ -568,11 +570,16 @@ func (a adminAPIHandlers) TemporaryAccountInfo(w http.ResponseWriter, r *http.Re
 	if sessionPolicy != nil {
 		stsAccountPolicy = *sessionPolicy
 	} else {
-		policiesNames, err := globalIAMSys.PolicyDBGet(stsAccount.ParentUser, false)
+		policiesNames, err := globalIAMSys.PolicyDBGet(stsAccount.ParentUser, stsAccount.Groups...)
 		if err != nil {
 			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 			return
 		}
+		if len(policiesNames) == 0 {
+			policySet, _ := args.GetPolicies(iamPolicyClaimNameOpenID())
+			policiesNames = policySet.ToSlice()
+		}
```

Comment:
I am not sure I understand this; this is an existing user, which means either it has an explicit policy or it will inherit the policy from the parent user;  if it has an explicit policy, this should be already handled in L569/571

Reply:
Not really, @vadmeste - there were always situations where we never saved parent-mapped policies in older releases. This is needed for backward compatibility. When customers upgrade from older deployments, we want their STS credentials to work until expired.

NEw STS will follow the parent-mapped policy approach.